src/client/Client.cc

```diff
@@ -1366,9 +1366,48 @@ void Client::dump_mds_requests(Formatter *f)
   }
 }
 
+class C_InodeRef {
+  Client *client;
+  Inode *in;
+public:
+  C_InodeRef(Client *c) : client(c), in(NULL) {}
+  C_InodeRef(Client *c, Inode *i) : client(c), in(i) {
+    if (in)
+      in->get();
+  }
+  C_InodeRef(const C_InodeRef& o) : client(o.client), in(o.in) {
+    if (in)
+      in->get();
+  }
+  ~C_InodeRef() {
+    if (in)
+      client->put_inode(in);
+  }
+  void swap(C_InodeRef& o) {
+    assert(client == o.client);
+    Inode *tmp = in;
+    in = o.in;
+    o.in = tmp;
+  }
+  C_InodeRef& operator=(Inode *in) {
+    assert(client);
+    C_InodeRef tmp(client, in);
+    swap(tmp);
+    return *this;
+  }
+  C_InodeRef& operator=(const C_InodeRef& o) {
+    C_InodeRef tmp(o);
+    swap(tmp);
+    return *this;
+  }
+  Inode *get() const { return in; }
```

Comment:
Why did you want to explicitly invoke a get() function instead of having an implicit conversion from C_InodeRef to Inode*?